Correción de bugs en el controlador y de las pruebas unitarias del POST.

rafamartinez21 opened the pull request:
Actualizado el controlador del POST y sus pruebas unitarias, optimización de código e implementación de comentarios.

A falta de revisión por SCRUM MASTER. #128

rafamartinez21 committed:
Correción de bugs en el controlador y de las pruebas unitarias del POST.

MarBauAB reviewed (changes requested):
postmerchcontroller: 
Linea 25: intenta usar solo Route("[action]"), el id puede darnos problemas en la siguiente

postmerch_test:
el expected que se define en la linea 197 debe definir todo, no lo tienes que asignar una vez se haya hecho el metodo en si, por lo que el objeto que se crea en la 226 no haría falta.
También las lineas:
            Assert.False(string.IsNullOrEmpty(actualDetail.CompraID));
            Assert.NotEqual("dummy-id", actualDetail.CompraID);

            // La fecha de compra debe ser reciente (dentro del ultimo minuto)
            Assert.True(actualDetail.FechaCompra > DateTime.Now.AddMinutes(-1));
son redundantes, no haría falta pues ya se comprueba con el assert.equal (o equivalent)

consejo: en el controller a la hora de definir la fecha pon datetime.today, pues con el .now pueden diferir en milisegundos y darte que no son las mismas

rafamartinez21 committed:
Realizadas las modificaciones:

Cambiado [Route("[action]/{id}")] por [Route("[action]")] en el GET.
Cambiado DateTime.Now por DateTime.Today en dos lugares.

Hemos eliminado el expectedDetail temporal, las verificaciones redundantes de compraID y FechaCompra, creamos un solo expectedDetails con valores del actualDetail y finalmente usamos assert.equal sin mas verificaciones.

MarBauAB reviewed (changes requested):
linea 185 de test: el expected tiene que crearse en bruto, no con datos en base al creado. el datetime de expected tendria que cambiarse a datetime.today y el compraid expcted al que se espere (en este caso, al no haber ninguno creado, el id esperado sería 1)

como en el model tienes que los ids de tus clases sean strings, te sugeriría que los cambiases a int para que sean autoincrementales y se puedan crear automaticamente y sea mas facil crearlos, asi en los casos de prueba en vez de tener que crear Guid.NewGuid(), podrías hacerlo simplemente que los ids sean 1,2,3 y así

rafamartinez21 committed:
Actualizados todos los archivos eliminación de id producto string por id producto int, omtización de codigo.

rafamartinez21 committed:
remove-migration

rafamartinez21 committed:
update-database.

MarBauAB reviewed (changes requested):
postmerch_test:

linea 192 y 193: definir los datos directamente, no en base a los del creado. Poner como id 1 y fechacompra datetime.today

rafamartinez21 committed:
DateTime.Today, // Usar la fecha real generada
   producto.PVP * 2 // Precio final calculado

cambiadas estas dos líneas de las pruebas unitarias del POST.

Actualizadas las pruebas unitarias del Get Select.

MarBauAB reviewed (approved):
<img width="1497" height="273" alt="imagen" src="https://github.com/user-attachments/assets/285e21ba-3c25-4474-83a0-bdaa860bfac8" />
todo correcto
